File: scripts/delete_users.ts
import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = process.env.VITE_SUPABASE_URL;
const SUPABASE_SERVICE_ROLE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!SUPABASE_URL || !SUPABASE_SERVICE_ROLE_KEY) {
  console.error("Missing VITE_SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY");
  process.exit(1);
}

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY);

async function deleteUsers() {
  const targetEmail = 'cgarness.ffl@gmail.com';
  let hasMore = true;
  let page = 1;
  let totalDeleted = 0;

  while (hasMore) {
    const { data: { users }, error } = await supabase.auth.admin.listUsers({ page, perPage: 1000 });
    
    if (error) {
      console.error('Error fetching users:', error);
      process.exit(1);
    }
    
    if (users.length === 0) {
      hasMore = false;
      break;
    }

    console.log(`Page ${page}: Found ${users.length} users.`);
    
    for (const user of users) {
      if (user.email === targetEmail) {
        console.log(`Skipping super admin: ${user.email} (${user.id})`);
        continue;
      }
      console.log(`Deleting: ${user.email || 'No email'} (${user.id})`);
      const { error: deleteError } = await supabase.auth.admin.deleteUser(user.id);
      if (deleteError) {
        console.error(`Failed to delete ${user.email}:`, deleteError.message);
      } else {
        totalDeleted++;
        console.log(`Successfully deleted ${user.email}`);
      }
    }
    
    if (users.length < 1000) {
      hasMore = false;
    } else {
      page++;
    }
  }
  
  console.log(`\nCompleted. Purged ${totalDeleted} users.`);
}

deleteUsers();

File: scripts/delete_users.py
import os
import urllib.request
import json
import ssl

def main():
    target_email = "cgarness.ffl@gmail.com"
    env_vars = {}
    with open(".env", "r") as f:
        for line in f:
            if "=" in line:
                key, val = line.strip().split("=", 1)
                env_vars[key] = val.strip('"').strip("'")
                
    url = env_vars.get("VITE_SUPABASE_URL")
    key = env_vars.get("SUPABASE_SERVICE_ROLE_KEY")
    
    if not url or not key:
        print("Missing URL or KEY")
        return

    admin_url = f"{url}/auth/v1/admin/users"
    headers = {
        "apikey": key,
        "Authorization": f"Bearer {key}"
    }
    
    ctx = ssl.create_default_context()
    ctx.check_hostname = False
    ctx.verify_mode = ssl.CERT_NONE

    req = urllib.request.Request(admin_url, headers=headers)
    try:
        with urllib.request.urlopen(req, context=ctx) as response:
            data = json.loads(response.read().decode())
    except Exception as e:
        print("Failed to fetch users:", e)
        return
        
    users = data.get("users", [])
    if isinstance(data, list):
        users = data
        
    print(f"Found {len(users)} users.")
    deleted = 0
    has_target = False
    for u in users:
        uid = u.get("id")
        email = u.get("email")
        if email == target_email:
            print(f"Skipping: {email} ({uid})")
            has_target = True
            continue
            
        print(f"Deleting: {email} ({uid})")
        del_req = urllib.request.Request(f"{admin_url}/{uid}", headers=headers, method="DELETE")
        try:
            with urllib.request.urlopen(del_req, context=ctx) as res:
                deleted += 1
                print("Deleted successfully")
        except Exception as e:
            print(f"Failed to delete {uid}: {e}")
            
    print(f"Completed. Purged {deleted} users.")
    
if __name__ == "__main__":
    main()
